coresdk/src/coresdk/database.h

```diff
@@ -1,60 +1,301 @@
+ *
+ * @returns Returns `true` if a database with the given `name` has
+ * has been loaded.
+ *
+ * @attribute class   database
+ * @attribute method  has_database
+ */
 bool has_database(string name);
 
+/**
+ * Retrieves a `database` that has been loaded into SplashKit.
+ *
+ * @param name The name of the database to return.
+ *
+ * @returns Returns the `database` that has been loaded with the specified
+ *          `name` using `open_database`.
+ *
+ * @attribute class   database
+ * @attribute method  database_named
+ */
 database database_named(string name);
 
+/**
+ * Calculates the number of row changed in the database at the last time
+ * a changing query was performed on the given database.
+ *
+ * @param db The database to check how many rows changed.
+ *
+ * @returns Returns the `int` which represents how many rows were changed
+ *          in the database.
+ *
+ * @attribute class   database
+ * @attribute method  rows_changed
+ */
 int rows_changed(database db);
 
+/**
+ * @brief Loads or creates, and returns a database
+ *
+ * The supplied `filename` is the name of the file of the database. The
+ * supplied `name` indicates the name to use to refer to this `database`.
+ * The `database` can then be retrieved by passing this `name` to
+ * the `database_named` function.
+ *
+ * @param name      The name used to refer to the database.
+ * @param filename  The filename used to locate the database to use.
+ *
+ * @returns A new `database` with the initialised values provided.
+ *
+ * @attribute class       database
+ * @attribute constructor true
+ */
 database open_database(string name, string filename);
 
+/**
+ * Returns the `query_result` for the query passed into `sql`.
+ *
+ * @param db The database to perform `sql` on.
+ *
+ * @param sql The sql statement to perform on `db`.
+ *
+ * @returns Returns the `query_result` which represents
+ * the result of perfoming `sql` on `db`.
+ *
+ * @attribute class   database
+ * @attribute method  run_sql_on_db
+ */
 query_result run_sql(database db, string sql);
 
+/**
+ * Returns the `query_result` for the query passed into `sql`.
+ *
+ * @param database_name The string name of the database to perform `sql` on.
+ *
+ * @param sql The sql statement to perform on the database at `database_name`.
+ *
+ * @returns Returns the `query_result` which represents
+ * the result of perfoming `sql` on the database at `database_name`.
+ *
+ * @attribute class   database
+ * @attribute method  run_sql_on_database_named
+ */
 query_result run_sql(string database_name, string sql);
 
+/**
+ * Frees all of the databases which have been loaded.
+ *
+ * @attribute class   database
+ * @attribute method  free_all_query_results
+ */
 void free_all_query_results();
 
+/**
+ * Releases the SplashKit resources associated with the query.
+ *
+ * @param query The `query_result` whose resources should be released.
+ *
+ * @attribute class       database
+ * @attribute self        query_result
```

Comment:
Can't use `@self` here as:

1. `@self` applies to parameter _names_, not _types_. In this case it should've been `query`.
2. The `@class` specified is a `database`, but the type of the `@self` is a `query_result`. Type mismatch would (should) occur in the parser.

It's a bit confusing.. we can go over it tomorrow :)